library/src/options.cxx

```diff
@@ -105,6 +105,48 @@ std::pair<std::string, unsigned int> options::getClosestOption(const std::string
   return ret;
 }
 
+//----------------------------------------------------------------------------
+template<>
+bool options::parse(const std::string& str)
+{
+  return options_tools::parse<bool>(str);
+}
+
+//----------------------------------------------------------------------------
+template<>
+int options::parse(const std::string& str)
+{
+  return options_tools::parse<int>(str);
+}
+
+//----------------------------------------------------------------------------
+template<>
+double options::parse(const std::string& str)
+{
+  return options_tools::parse<double>(str);
+}
+
+//----------------------------------------------------------------------------
+template<>
+f3d::ratio_t options::parse(const std::string& str)
+{
+  return options_tools::parse<f3d::ratio_t>(str);
+}
+
+//----------------------------------------------------------------------------
+template<>
+std::string options::parse(const std::string& str)
+{
+  return options_tools::parse<std::string>(str);
+}
+
+//----------------------------------------------------------------------------
+template<>
+std::vector<double> options::parse(const std::string& str)
+{
+  return options_tools::parse<std::vector<double>>(str);
+}
+
```

Comment:
what about doing
```cpp
template<typename T>
T options::parse(const std::string& str)
{
  return options_tools::parse<T>(str);
}

template bool options::parse<bool>(const std::string& str);
template int options::parse<int>(const std::string& str);
// ...
```

that way, you have a oneliner when you need to add a type.

Reply:
You could even do something like:
```cpp
#define F3D_DECL_TYPE(TYPE) template TYPE options::parse<TYPE>(const std::string& str)
F3D_DECL_TYPE(bool);
F3D_DECL_TYPE(int);
// ...
```
Yes I know, macros are bad but in this case it makes sense.

Reply:
fair point.